applications/luci-app-firewall/luasrc/model/cbi/firewall/zones.lua

```diff
@@ -32,6 +32,18 @@ for i, v in ipairs(p) do
 	v:value("ACCEPT", translate("accept"))
 end
 
+s:option(DummyValue, "offload_advice",
+	translate("Routing/NAT Offloading"),
+	translate("Exprimental feature. Requires kernel 4.14 and is not fully compatible with QoS/SQM."))
+
+o = s:option(Flag, "flow_offloading", translate("Software flow offloading"),
+	translate("Requires kernel 4.14"))
+o.optional = true
+
+o = s:option(Flag, "flow_offloading_hw", translate("Hardware flow offloading"),
```

Comment:
Here we could also check for "/sys/module/nf_flow_table_hw" or something else that indicates availability

Reply:
fw3 currently does this:
https://git.openwrt.org/?p=project/firewall3.git;a=blob;f=defaults.c;h=11fbf0dd8f74a51e4b2d95fd0bb7e742d65cd8f4;hb=30463d08056db8f03d17c0569581aef96418c38d#l85

Reply:
I´ve added this and is working fine:
```
local fo = io.open("/sys/module/nf_flow_table/initstate", "r")
local foh = io.open("/sys/module/nf_flow_table_hw/initstate", "r")
... option flow offload
if fo ~= nil then
... option flow offload hw
if foh ~= nil then
...
end
end
```
Is that to much processing in the lua scripts?
The error message from fw3 is only printed on cli and user did not get a feedback if it is properly enabled in luci...

Reply:
I will follow jow's example from firewall3 code itself.

My own ipq806x/R7800 has /sys/module/nf_flow_table_hw/initstate although there is no hardware support, yet, so that file does not indicate anything.

Reply:
Sorry, forgot to say that i don't konw if that files indicate availability... Tested it only on mt7621 and ar71xx device.